lms/services/lis_result_sourcedid.py

```diff
@@ -0,0 +1,57 @@
+from lms.models import LISResultSourcedId
+
+__all__ = ["LISResultSourcedIdService"]
+
+
+class LISResultSourcedIdService:
+    """Methods for interacting with LISResultSourcedId records."""
+
+    def __init__(self, _context, request):
+        self._db = request.db
+
+    def upsert(self, validated_attrs):
```

Comment:
This method doesn't handle any Exceptions whatsoever, assuming that the `validated_attrs` have been, well, validated. Let me know if there are things I should be doing here...

Reply:
Looks fine. The service should handle any DB errors that come up---either by handling them or by raising documented exceptions. But I don't think there are any DB errors possible here. It's not a problem if no record exists, or if a conflicting record already exists, for example. So looks good to me